tests/integration/test_backup_restore_on_cluster/test.py

```diff
@@ -1199,3 +1199,34 @@ def test_shutdown_waits_for_backup():
     node1.query(f"RESTORE TABLE tbl ON CLUSTER 'cluster' FROM {backup_name}")
     node1.query("SYSTEM SYNC REPLICA tbl")
     assert node1.query("SELECT * FROM tbl ORDER BY x") == TSV([3, 5])
+
+
+def test_replicated_table_after_alters():
+    node1.query(
+        "CREATE TABLE tbl ON CLUSTER 'cluster' ("
+        "x Int32"
+        ") ENGINE=ReplicatedMergeTree('/clickhouse/tables/tbl/', '{replica}')"
+        "ORDER BY x"
+    )
+
+    node1.query("INSERT INTO tbl VALUES (1)")
+    node1.query("ALTER TABLE tbl ADD COLUMN y Int32")
+    node1.query("INSERT INTO tbl VALUES (2, 20)")
+    node1.query("ALTER TABLE tbl ADD COLUMN z Int32")
+    node1.query("INSERT INTO tbl VALUES (3, 30, 300)")
+
+    backup_name = new_backup_name()
+
+    node1.query(f"BACKUP TABLE tbl ON CLUSTER 'cluster' TO {backup_name}")
+    node1.query("DROP TABLE tbl ON CLUSTER 'cluster' SYNC")
+    node1.query(f"RESTORE TABLE tbl ON CLUSTER 'cluster' FROM {backup_name}")
+
+    node1.query("OPTIMIZE TABLE tbl FINAL")
```

Comment:
Without this PR's correction `OPTIMIZE TABLE` will stuck forever with repeating those two lines in the logs:
```
2024.12.23 21:12:46.330342 [ 588 ] {} <Trace> ceabdacf-89d8-464d-8a5e-63c19d5aa2b0::all_0_2_1 (MergeFromLogEntryTask): Executing log entry to merge parts all_0_0_0, all_1_1_0, all_2_2_0 to all_0_2_1
2024.12.23 21:12:46.330361 [ 588 ] {} <Debug> ceabdacf-89d8-464d-8a5e-63c19d5aa2b0::all_0_2_1 (MergeFromLogEntryTask): Source part metadata version 1 is newer then the table metadata version 0. ALTER_METADATA is still in progress.
```